### PR TITLE
Change regex that validates username to allow non-Latin characters

### PR DESCRIPTION
**Pull request by jonboiser**
addresses #1876 

Regex does not accept spaces or any punctuation on US keyboard. To blacklist punctuation from a larger set of keyboards would require a more comprehensive blacklist.

https://stackoverflow.com/questions/7576945/javascript-regular-expression-for-punctuation-international

screen shot 2017-08-30 at 3 47 20 pm
screen shot 2017-08-30 at 3 47 15 pm

### CODE REVIEW

File: kolibri/plugins/setup_wizard/assets/src/views/index.vue [hunk 1]
```diff
@@ -154,7 +153,8 @@
         return !!this.username;
       },
       usernameValidityCheck() {
-        return /^\w+$/g.test(this.username);
+        const hasPuncRe = /[\s`~!@#$%^&*()-+={}\[\]\|\\\/:;"'<>,\.\?]/; // eslint-disable-line
```
Comment (rtibbles) [inline kolibri/plugins/setup_wizard/assets/src/views/index.vue]: We should probably abstract the username validity functionality into core, using this, and then use it in all places that usernames are created or can be edited.
https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/management/assets/src/views/user-page/user-create-modal.vue#L120
https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/management/assets/src/views/user-page/user-edit-modal.vue#L22
https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/user/assets/src/views/profile-page/index.vue#L25

Comment (christianmemije) [reply kolibri/plugins/setup_wizard/assets/src/views/index.vue]: Btw, I added username validation in all those places, with the exception of the user-edit-modal, which is up next, in this pr https://github.com/learningequality/kolibri/pull/2005

File: kolibri/plugins/setup_wizard/assets/src/views/index.vue [hunk 2]
```diff
@@ -138,7 +138,8 @@
     },
     computed: {
       usernameValidityCheck() {
-        return /^\w+$/g.test(this.username);
+        const hasPuncRe = /[\s`~!@#$%^&*()-+={}\[\]\|\\\/:;"'<>,\.\?]/; // eslint-disable-line
```
Comment (indirectlylit) [inline kolibri/plugins/setup_wizard/assets/src/views/index.vue:141]: image

https://xkcd.com/1638/

**Comment by christianmemije** [comment]
@jonboiser Note that there are many other places where we validate the username, so those regexs need to also be updated. Might be good time to dry up this validation check.

**Pull request merged**